core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/MetricCategory.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.metrics.internal.SdkMetric;
+
+/**
+ * A enum class representing the different types of metric categories in the SDK.
+ *
+ * A metric can be tagged with multiple categories. Clients can enable/disable metric collection
+ * at a {@link MetricCategory} level.
+ *
+ * @see SdkMetric
+ */
+@SdkPublicApi
+public enum MetricCategory {
+
+    /**
+     * All metrics defined by the SDK are classified under this category at a minimum. If the metrics feature is enabled
+     * but the category to collect is not, only metrics that are classified under this category are collected by the SDK
+     */
```

Comment:
Can we add links to all of these to the metrics that they will collect?

Reply:
Right now, all the metrics in the SdkMetric class belong to `DEFAULT` category. Will update after more metrics are added